source/JsMaterialX/test/document.spec.js

```diff
@@ -12,186 +12,137 @@ describe('Build Document', () => {
     function expectError(type, cb) {
         try {
             cb();
+            throw new Error('Expected function to throw!');
         } catch (exceptionPtr) {
             const message = mx.getExceptionMessage(exceptionPtr);
             expect(message.indexOf(type) !== -1).to.be.true;
         }
     }
 
     let nodeGraph;
-    it('Create a node graph with constant and image sources.', () => {
+    it('Build document', () => {
```

Comment:
This have been many small tests that were depending on each other. They have been consolidated into one large test that is now equivalent to the python one.